doc/users/interactive_guide.rst

```diff
@@ -0,0 +1,86 @@
+.. _plotting-guide-interactive:
+
+*******************
+Interactive Figures
+*******************
+
+One of the most powerful uses of matplotlib is interactive
+figures.  At the most basic matplotlib has the ability to zoom and pan
+a figure to inspect your data, however there is also a full mouse and
+keyboard event handling system to enable building sophisticated interactive
+graphs.
+
+This page is meant to be a rapid introduction to the relevant details of
+integrating the matplotlib with a GUI event loop.
+
+
+The GUI event loop
+------------------
+
+To handle asynchronous user input every GUI framework has an event
+loop.  At the most basic this is a stack that can have events to be
+processed.  In order for the GUI to be responsive this loop must be
+run.  To manage this in python there are two basic methods:
+
+1. let the GUI main loop block the python process
+2. intermittently run the GUI loop for a period of time
+
+Going with option 1 is going down the route of writing a bespoke GUI
+application.  In this case, commonly refereed to as 'embedding', the
+GUI event loop is running the show and you should not use the `pyplot`
+layer.  Doing anything short of writing a full GUI application
+requires option 2.
+
+The python capi provides a hook, `PyOS_InputHook`, to register a
+function to be run "The function will be called when Python's
+interpreter prompt is about to become idle and wait for user input
+from the terminal.".  This hook can be used to integrate a second
```

Comment:
Why "second event loop"? What would be the first event loop?

Reply:
The python prompt itself.